Shoko.Server/API/v3/Controllers/DashboardController.cs

```diff
@@ -240,6 +240,46 @@ public List<Series> GetRecentlyAddedSeries([FromQuery] int pageSize = 20, [FromQ
                 .ToList();
         }
 
+        /// <summary>
+        /// Get a list of the episodes to continue watching (soon-to-be) in recently watched order.
+        /// </summary>
+        /// <param name="pageSize">Limits the number of results per page. Set to 0 to disable the limit.</param>
+        /// <param name="page">Page number.</param>
+        /// <returns></returns>
+        [HttpGet("ContinueWatchingEpisodes")]
+        public List<Dashboard.EpisodeDetails> GetContinueWatchingEpisodes([FromQuery] int pageSize = 20, [FromQuery] int page = 0)
+        {
+            var user = HttpContext.GetUser();
+            var episodeList = RepoFactory.VideoLocalUser.Cache.Values
+                .Where(userData => userData.JMMUserID == user.JMMUserID && userData.WatchedDate == null && userData.ResumePosition != 0)
+                .OrderByDescending(userData => userData.DateTimeUpdated)
+                .Select(userData => RepoFactory.VideoLocal.GetByID(userData.VideoLocalID))
+                .Where(file => file != null)
+                .Select(file => file.EpisodeCrossRefs.OrderBy(xref => xref.EpisodeOrder).ThenBy(xref => xref.Percentage).FirstOrDefault())
+                .Where(xref => xref != null)
+                .Select(xref => RepoFactory.AnimeEpisode.GetByAniDBEpisodeID(xref.EpisodeID))
+                .Where(episode => episode != null)
+                .DistinctBy(episode => episode.AnimeSeriesID)
+                .ToList();
+            var seriesDict = episodeList
+                .Select(episode => episode.GetAnimeSeries())
+                .Where(series => series != null)
+                .ToDictionary(series => series.AnimeSeriesID);
+
+            if (pageSize <= 0)
+                return episodeList
+                    .Where(episode => user.AllowedSeries(seriesDict[episode.AnimeSeriesID]))
+                    .Select(a => new Dashboard.EpisodeDetails(a.AniDB_Episode, seriesDict[a.AnimeSeriesID].GetAnime(), seriesDict[a.AnimeSeriesID]))
+                    .ToList();
+            if (page <= 0) page = 0;

```

Comment:
This should be part of a validation rule

Reply:
I didn't know the range validation rule existed until earlier today. Will convert this now, then the other endpoints using the same pattern later today (in another PR).